relay-profiling/src/lib.rs

```diff
@@ -275,41 +283,72 @@ pub fn expand_profile(
     }
 }
 
-pub fn expand_profile_chunk(
-    payload: &[u8],
-    client_ip: Option<IpAddr>,
-    filter_settings: &ProjectFiltersConfig,
-    global_config: &GlobalConfig,
-) -> Result<Vec<u8>, ProfileError> {
-    let profile = match minimal_profile_from_json(payload) {
-        Ok(profile) => profile,
-        Err(err) => {
-            relay_log::warn!(
-                error = &err as &dyn Error,
-                from = "minimal",
-                "invalid profile chunk",
-            );
-            return Err(ProfileError::InvalidJson(err));
+/// Intermediate type for all processing on a profile chunk.
+pub struct ProfileChunk {
+    profile: MinimalProfile,
+    payload: Bytes,
+}
+
+impl ProfileChunk {
+    /// Parses a new [`Self`] from raw bytes.
+    pub fn new(payload: Bytes) -> Result<Self, ProfileError> {
+        match minimal_profile_from_json(&payload) {
+            Ok(profile) => Ok(Self { profile, payload }),
+            Err(err) => {
+                relay_log::debug!(
+                    error = &err as &dyn Error,
+                    from = "minimal",
+                    "invalid profile chunk",
+                );
+                Err(ProfileError::InvalidJson(err))
+            }
         }
-    };
+    }
 
-    if let Err(filter_stat_key) = relay_filter::should_filter(
-        &profile,
-        client_ip,
-        filter_settings,
-        global_config.filters(),
-    ) {
-        return Err(ProfileError::Filtered(filter_stat_key));
+    /// Returns the [`ProfileType`] this chunk belongs to.
+    ///
+    /// The profile type is currently determined based on the contained profile
+    /// platform. It determines the data category this profile chunk belongs to.
+    ///
+    /// This needs to be synchronized with the implementation in Sentry:
```

Comment:
If we can add the `profile_type` in the payload before we send it, this would become the source of truth.

Reply:
I think we can use add it to the Kafka message https://github.com/getsentry/relay/blob/master/relay-server/src/services/store.rs#L1035-L1041 and the worker will handle it, no need to deserialize/serialize again.

Reply:
Sure, that works, let's figure that out in a follow up where we wanna put it (payload / kafka header) and how it should look?

Reply:
I would add it to the payload and not the header, it's easier to manipulate later on.